aqt/installer.py

```diff
@@ -472,7 +481,7 @@ def run_tool(self, args):
             )
         )
 
-    def run_list(self, args):
+    def run_list_old(self, args):
```

Comment:
@miurahr, when you asked me to extend `aqt list`, I wasn't sure how to preserve the original implementation. You asked for:

> When user specify `category host target`, aqt run the proposed feature and show versions and packages.
> When user specify `category host target version` then list packages by reading Updates.xml.

You provided an expected output for the `category host target version` command, which is a list of packages. The existing implementation of `aqt list` prints a table that does not match the expected output that you asked for. I implemented `helper.get_packages_for_version`, which prints the list of packages that you asked for, and removed the existing implementation.

Please let me know if I misunderstood what you meant.

Reply:
Ok, I can remove the existing implementation. I will also remove the aqt.archives.PackagesList class. It is only used in Cli.run_list, and when I remove that function, it becomes dead code.

Reply:
I only have implementations for options 1 and 4. I will avoid removing aqt.archives.PackagesList; there must be a way to reuse it for options 2 and 3.

I think we need a well-defined interface for accessing options 1-4. I don't like the idea of reusing positional arguments to mean different things; this makes the help text difficult to read and the interface unclear. Consider this possible interface:
```
$ list <category> <host OS> <target>   # 1. Print a list of all tools/versions of Qt available for category, os, target
$ list <Qt version> <host OS>          # 2. Print a list of all targets for a version of Qt
$ list <Qt version> <host OS> <target> # 3. Print a list of all architectures available for Qt version, os, target
$ list <Qt version> <host OS> <target> --modules # 4. Print all modules for (Qt, OS, target) tuple
```
This could work, but the first positional argument is confusing. In the help text, you would have to call it something like `QT_VERSION_OR_CATEGORY`, and the help text would have a lot of explaining to do.

This is an interface that I would prefer, as a user. It is more verbose, but it's also more consistent.
```
$ list <category> <host OS> <target>   # 1. Print a list of all tools/versions of Qt available for category, os, target
$ list <category> <host OS> --targets <Qt version>       # 2. Print a list of all targets for a version of Qt
$ list <category> <host OS> <target> --arch <Qt version> # 3. Print a list of all architectures available for Qt version, os, target
$ list <category> <host OS> <target> --modules <Qt version> # 4. Print all modules for (Qt, OS, target) tuple
```

Another possibility is to turn option 1 into separate subcommands. This option is less verbose and easier to understand:

```
$ list-tools <host OS> <target> # 1.1. Print a list of all tools available for os, target
$ list-ver <host OS> <target>   # 1.2. Print a list of all versions of Qt available for category, os, target. 
                                # There could be flags for filtering by major and minor versions.
$ list <Qt version> <host OS>          # 2. Print a list of all targets for a version of Qt
$ list <Qt version> <host OS> <target> # 3. Print a list of all architectures available for Qt version, os, target
$ list <Qt version> <host OS> <target> --modules # 4. Print all modules for (Qt, OS, target) tuple
```

These are the best suggestions I can come up with right now. There are many other ways to solve this problem; maybe you can come up with something better?

Reply:
Thank you for detailed examination and design considerations! 🥇 

> This is an interface that I would prefer, as a user. It is more verbose, but it's also more consistent.

I prefer a middle No. 2️⃣ , too. Could you proceed your design? 🚀 

```
$ list <category> <host OS> <target>   # 1. Print all tools/versions of Qt available for category, os, target
$ list <category> <host OS> --targets <Qt version>       # 2. Print all targets for a version of Qt
$ list <category> <host OS> <target> --arch <Qt version> # 3. Print all architectures for Qt version, os, target
$ list <category> <host OS> <target> --modules <Qt version> # 4. Print all modules for (Qt, OS, target) tuple
```

Reply:
Ok, number 2 it is!